R/parameters_selection.R

```diff
@@ -0,0 +1,141 @@
+#' Parameters Selection
+#'
+#' Automated selection of the 'best' parameters. For frequentist simple GLMs, it performes an AIC-based stepwise selection. For Bayesian models, it uses the \code{projpred} package.
+#'
+#' @param model A statistical model.
+#' @param ... Arguments passed to or from other methods.
+#'
+#'
+#' @examples
+#' model <- lm(mpg ~ ., data = mtcars)
+#' parameters_selection(model)
+#'
+#' model <- lm(mpg ~ cyl * disp * hp * wt , data = mtcars)
+#' parameters_selection(model, cross_validation = FALSE)
+#'
+#' \dontrun{
+#' # lme4 -------------------------------------------
+#' library(lme4)
+#' model <- lmer(Sepal.Width ~ Sepal.Length * Petal.Width * Petal.Length + (1|Species), data = iris)
+#' parameters_selection(model)
+#'
+#' # rstanarm -------------------------------------------
+#' library(rstanarm)
+#' model <- stan_glm(mpg ~ ., data = mtcars)
+#' parameters_selection(model, cross_validation = TRUE)
+#'
+#' model <- stan_glm(mpg ~ cyl * disp * hp * wt , data = mtcars)
+#' parameters_selection(model, cross_validation = FALSE)
+#' }
+#'
+#'
+#' @export
+parameters_selection <- function(model, ...){
+  UseMethod("parameters_selection")
+}
+
+
+
+#' @rdname parameters_selection
+#' @importFrom stats step
+#' @export
+parameters_selection.lm <- function(model, ...){
+  junk <- capture.output(best <- step(model, ...))
+
+  parameters <- names(best$coefficients)
+  formula <- .reconstruct_formula(parameters)
+  formula
+}
+
+
+
+
+
+
+
+
+
+#' @rdname parameters_selection
+#' @export
+parameters_selection.merMod <- function(model, ...){
+  if (!requireNamespace("cAIC4", quietly = TRUE)) {
+    stop("Package `cAIC4` required. Please install it by running `install.packages(cAIC4)`.", call. = FALSE)
+  }
+
+  best <- cAIC4::stepcAIC(model)
+  # TODO: Need to improve reconstruct formula to also account for random effects and all
+  parameters <- insight::find_parameters(best$finalModel, effects = "fixed", flatten = TRUE)
```

Comment:
Is here something we need to address in insight?

Reply:
This depends on if/what we want to do with the formula reconstruction

Reply:
Not sure in which way you need to reconstruct a formula, I probably need to look a bit closer at the code.